CompositionRoot/B.swift

```diff
@@ -1,17 +1,17 @@
-protocol BPrototol {
+protocol BProtocol {
     func bMethod()
 }
 
-class B: BPrototol {
+class B: BProtocol {
     
-    let applicationContext: ApplicationContext
+    let cFactory: CFactoryProtocol
     
-    init(applicationContext: ApplicationContext) {
-        self.applicationContext = applicationContext
+    init(cFactory: CFactoryProtocol) {
```

Comment:
`Composition Root` pattern 
1. remove the dependency of `B` from `ApplicationContext`
2. Remove the dependency on specific class `C` and make it depend on abstract interface `CFactoryProtocol`.  OOP SOLID principles encourage depending on abstract interface and discourage depending on specific class. 
3. removing dependency on concrete class C makes unit test easier. 
4. decouple class B and class C.

These are the benefits `Composition Root` pattern bring to us.